lib/api/apiUtils/bucket/parseWhere.js

```diff
@@ -40,6 +40,35 @@ const exprMapper = {
     'LIKE': '$regex',
 };
 
+function parseLikeExpression(value, operator) {
+    const expr = exprMapper[operator];
+    let obj;
+    let pattern;
+
+    const regex = value.split('/');
+    if (regex.length < 3) {
+        pattern = value;
+        obj = { [expr]: value };
+    } else {
+        pattern = regex.slice(1, regex.length - 1).join('/');
+        const regexOpt = regex[regex.length - 1];
+        const invalidOpts = regexOpt.split('').some(
+            opt => !['i', 'x', 'm', 's'].includes(opt));
+        const dupOpts = regexOpt.split('').some(
+            (opt, ind, arr) => arr.lastIndexOf(opt) !== ind);
+        if (invalidOpts || dupOpts) {
+            throw new Error(`Invalid regex options "${regexOpt}"`);
```

Comment:
I recommend returning the error and in the caller method you can check for `instanceof Error.`